mi/dataset/parser/suna.py

```diff
@@ -209,7 +209,7 @@ def get_records(self, num_records_requested=1):
 
             # Iterate through the particles returned, and pop them off from the beginning of the record
             # buffer to the end
-            while len(particles_to_return) is not num_records_requested and len(self._record_buffer) > 0:
+            while len(particles_to_return) is not num_records_requested and len(self._record_buffer):
```

Comment:
`len(self._record_buffer)` is not necessary - you can just use `self._record_buffer` which will return true if there is anything in the buffer - magic!